api/src/pcapi/scripts/offer/copy_offer_from_venue.py

```diff
@@ -0,0 +1,243 @@
+from pcapi.flask_app import app
+from pcapi.models.offer_mixin import OfferValidationType
+from pcapi.routes.backoffice.filters import format_date
+from pcapi.repository import transaction
+from pcapi.models import db
+
+OFFERS_TO_COPY_BEFORE_CSV = "offers_to_copy_before.csv"
+OFFERS_TO_COPY_AFTER_CSV = "offers_to_copy_after.csv"
+
+OFFERS_COPIED_CSV = "offers_copied.csv"
+
+STOCKS_TO_COPY_CSV = "stocks_to_copy.csv"
+STOCKS_COPIED_CSV = "stocks_copied.csv"
+
+# local
+FROM_VENUE_ID = 4
+TO_VENUE_ID = 2
+
+# testing
+# FROM_VENUE_ID = 10947
+# TO_VENUE_ID = 10948
+
+# staging and prod
+# FROM_VENUE_ID=2525
+# TO_VENUE_ID=92072
+
+EXTRACT_FOLDER = "/tmp/"
+
+
+def delete_files_csv(list_filenames: list[str]):
+    for filename in list_filenames:
+        file = os.path.join(EXTRACT_FOLDER, filename)
+        # If file exists, delete it.
+        if os.path.isfile(file):
+            os.remove(file)
+
+
+def _copy_offers(dryrun: bool):
+    description = f"copy offer from venue {FROM_VENUE_ID} to venue {TO_VENUE_ID}"
+    print(description)
+
+    offers_to_copy = (
+        offer_models.Offer.query.join(offer_models.Stock)
+        .filter(offer_models.Offer.venueId == FROM_VENUE_ID)
+        .filter(offer_models.Offer.subcategoryId == subcategories.MATERIEL_ART_CREATIF.id)
+        .filter(offer_models.Offer.validation == offer_mixin.OfferValidationStatus.APPROVED)
+        .filter(sa.or_(offer_models.Stock.remainingQuantity.is_(None), offer_models.Stock.remainingQuantity > 0))
+        .all()
+    )
+
+    print(f"{len(offers_to_copy)} offers to copy")
+
+    _print_offers_csv(offers_to_copy, OFFERS_TO_COPY_BEFORE_CSV)
+
+    if offers_to_copy:
+        stocks_copied = []
+        offers_copied = []
+        all_stocks_to_copy = []
+        i = 0
+        with transaction():
+            for offer_to_copy in offers_to_copy:
+                offer_copied = duplicate_object(offer_to_copy, offer_models.Offer)
+                offer_copied.venueId = TO_VENUE_ID
+                db.session.add(offer_copied)
+
+                offers_copied.append(offer_copied)
+
+                stocks_to_copy: list[offer_models.Stock] = offer_models.Stock.query.filter(
+                    offer_models.Stock.offerId == offer_to_copy.id
+                ).all()
+
+                # print(f"{len(stocks_to_copy)} stocks for offer {offer_to_copy.id}")
+                all_stocks_to_copy.extend(stocks_to_copy)
+
+                price_category_by_offer = {}
+
+                for stock_to_copy in stocks_to_copy:
+                    stock_copied: offer_models.Stock = duplicate_object(stock_to_copy, offer_models.Stock)
+                    stock_copied.quantity = (
+                        None
+                        if stock_to_copy.quantity is None
+                        else stock_to_copy.quantity - stock_to_copy.dnBookedQuantity
+                    )
+                    stock_copied.dnBookedQuantity = 0
+                    stock_copied.offerId = offer_copied.id
```

Comment:
En fait comme tu fais `db.session.add` de l'offre, mais que tu n'as encore `commit`, a priori ça ne sait pas trouver l'`id`. Je pense que ça devrait marcher ainsi, à tester : 
```suggestion
                    stock_copied.offer = offer_copied
```